secure/key-manager/src/lib.rs

```diff
@@ -172,11 +202,33 @@ where
         Ok(new_key)
     }
 
-    pub fn evaluate_status(&self) -> Result<Action, Error> {
-        // If this is inconsistent, then we are likely waiting on a reconfiguration.
+    /// Ensures that the libra_timestamp() value registered on-chain is strictly monotonically
+    /// increasing.
+    fn ensure_timestamp_progress(&mut self) -> Result<(), Error> {
+        let current_libra_timestamp = self.libra.libra_timestamp()?;
+        if current_libra_timestamp <= self.last_checked_libra_timestamp {
```

Comment:
I'm slightly concerned by this
1) what if we query rapidly -- I guess this could force some sort of backoff, but it could also confuse us
2) it doesn't address that libra time is moving at a rate that makes sense -- but maybe that doesn't matter for key manager

Reply:
Yeah, I had the same concerns when writing this:

1. This could definitely happen, hence why I tried to make the comment super obvious. One way of avoiding it is to make the progress check optional, however, if we do want to ensure liveness during the evaluate_status() function, this seems the only real way to enforce it (i.e., to return an error). Ideally, it would nice to return an error that differentiates between a Liveness problem and a 'calling too frequently' problem, but that means answering a much harder question 😄

2. I agree, but it's not immediately clear to me how exactly we do this? I mean, what is a rate that makes sense? For now, I figured, we could at least ensure that the time is moving forward..